src/shims/foreign_items.rs

```diff
@@ -124,8 +130,24 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriEvalContextExt<'mir, 'tcx
 
         // First: functions that diverge.
         match link_name {
-            "__rust_start_panic" | "panic_impl" => {
-                throw_unsup_format!("the evaluated program panicked");
+            // Note that this matches calls to the *foreign* item "__rust_start_panic" -
+            // that is, calls `extern "Rust" { fn __rust_start_panic(...) }`
+            // We forward this to the underlying *implementation* in "libpanic_unwind"
+            "__rust_start_panic" => {
+                let start_panic_instance = this.resolve_path(&["panic_unwind", "__rust_start_panic"])?;
+                return Ok(Some(this.load_mir(start_panic_instance.def, None)?));
+            }
+
+            // Normally, this gets resolved to the '#[panic_handler]` function
+            // during compilation. We manually forward to the panic_impl lang item,
+            // which corresponds to the function with the `#[panic_handler]` attribute
```

Comment:
"Normally" sounds like there's something abnormal here, but I think this is the same thing as with `__rust_start_panic` above, isn't it? This matches calls to a foreign item (which is what libcore calls to forward panics to the actual panic impl), and we forward to the *implementation* which is a lang item (defined by `#[panic_handler]`).

Reply:
By 'Normally', I was referring to a normal Rust compilation. I'll make that explicit.

Reply:
The thing is, this is exactly the same as the case just above. So why is the comment so different?